server/test/com/cloud/api/query/dao/GenericDaoBaseWithTagInformationBaseTest.java

```diff
@@ -0,0 +1,59 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.api.query.dao;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+
+import org.apache.cloudstack.api.BaseResponseWithTagInformation;
+import org.apache.cloudstack.api.response.ResourceTagResponse;
+import org.powermock.api.mockito.PowerMockito;
+
+import com.cloud.api.ApiDBUtils;
+import com.cloud.api.query.vo.BaseViewWithTagInformationVO;
+
+public abstract class GenericDaoBaseWithTagInformationBaseTest<T extends BaseViewWithTagInformationVO,
+                                                                ID extends Serializable,
+                                                                Z extends BaseResponseWithTagInformation> {
+
+    public void prepareSetup(String tagKey, String tagValue){
+        PowerMockito.spy(ApiDBUtils.class);
+        PowerMockito.stub(PowerMockito.method(ApiDBUtils.class, "newResourceTagResponse")).toReturn(getResourceTagResponse(tagKey, tagValue));
+    }
+
+    private ResourceTagResponse getResourceTagResponse(String tagKey, String tagValue){
+        ResourceTagResponse tagResponse = new ResourceTagResponse();
+        tagResponse.setKey(tagKey);
+        tagResponse.setValue(tagValue);
+        return tagResponse;
+    }
+
+    public void testUpdateTagInformation(GenericDaoBaseWithTagInformation<T, ID, Z> dao, T baseView, Z baseResponse,
```

Comment:
I added a new test case and changed values to base test, so there's need to pass them